src/grafana_agent.py

```diff
@@ -615,6 +615,7 @@ def _generate_config(self) -> Dict[str, Any]:
                 "wal_directory": "/tmp/agent/data",
                 "global": {
                     "scrape_timeout": self.model.config.get("global_scrape_timeout"),
+                    "scrape_interval": self.model.config.get("scrape_interval"),
```

Comment:
```suggestion
                    "scrape_interval": self.model.config.get("scrape_interval"),
                    "scrape_interval": self.model.config.get("global_scrape_interval")
```

Reply:
`yaml.dump()` would conver the `None` to a `null`, making the configuration invalid I'd assume